README.md

```diff
@@ -12,34 +26,34 @@ Go clients for talking to a [kubernetes](http://kubernetes.io/) cluster.
 
 ### Versioning
 
-`client-go` will follow [semver](http://semver.org/). We will not make backwards-incompatible changes without incrementing the major version number. A branch will be created for each increase in the major version number (first digit of the version) or minor version number (second digit of the version). A new tag will be created for each increase in the patch version number (third digit of the version).
+`client-go` version numbers are unrelated to Kubernetes version numbers. Please see the [compatibility matrix](#compatibility-matrix) for the compatible Kubernetes clusters.
 
-The master branch will track the head in the main Kubernetes repo and accumulating changes. We will bump the client-go version when:
-* A new major (very rare) or a new minor version is released in the Kubernetes main repository.
-* A feature in the master branch is popular and users want it in a stable branch. 
+`client-go` follows [semver](http://semver.org/). We will not make backwards-incompatible changes without incrementing the major version number. A change is backwards-incompatible either if it *i)* changes the public interfaces of `client-go`, or *ii)* makes `client-go` incompatible with otherwise supported versions of Kubernetes clusters.
 
-Depending on the nature of the changes accumulated when we bump the version, we will bump version number according to semver.
+We will create a new branch for each increment in the major version number or minor version number. We will create a new tag for each increment in the patch version number. See [semver](http://semver.org/) for definitions of major, minor, and patch.
 
-Note that these are the versions *of the client library itself*, not the versions of the Kubernetes clusters which the library will talk to. 
+The master branch will track the head in the main Kubernetes repo and accumulating changes. We will make a new client-go major/minor/patch version when:
+* A new major (very rare) or a new minor version is released in the Kubernetes main repository.
+* A feature or a bug fix in the master branch is popular and users want it in a stable branch or with a stable tag. 
 
-#### Compatibility: Your code <-> client-go
+#### Compatibility: your code <-> client-go
 
-`client-go` will follow [semver](http://semver.org/), so your code keeps compatible until the major version of client-go gets bumped.
+`client-go` follows [semver](http://semver.org/), so until the major version of client-go gets increased, your code keeps compilable and keeps working with explicitly supported versions of Kubernetes clusters.
```

Comment:
"[...] your code will compile and will continue to work with explicitly supported [...]" reads better.